base/resources/scripts/languagecheck.js

```diff
@@ -0,0 +1,40 @@
+/* Nodejs to set up a server on 8081 to modify the argv.json file
+   used by in-browser VSCode. Uses the accept-language header.
+   Also used by jupyter to determine language.
+ */
+var http = require("http");
+var fs = require("fs");
+
+http
+  .createServer(function (req, res) {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    var lang = req.headers["accept-language"];
+    lang ? lang : (lang = "en");
+    lang = lang.substring(0, 2);
+
+    lang === "fr" || lang === "en" ? lang : (lang = "en");
+
+    // READ THE FILE AND COMPARE TO LANG IF IT CHANGES DO THE FOLLOWING, SYNC to not write to it as well
+    var file = fs.readFileSync(
+      "/home/jovyan/.config/Code/User/argv.json",
+      "utf-8"
+    ); // must specify encoding to return string
+    var browserLang = '"locale":"' + lang + '"'; //search using this, "locale":"en" if browserLang not in file then change
+
+    if (file.indexOf(browserLang) == -1) {
```

Comment:
You can often invert a condition like this to switch to a "guard" form. It will limit the amount of code sitting under the condition and make it a bit of an easier read.

```js
if (file.index(browserLang) >= 0) {
  return res.end(...) // The `return` ensures we don't accidentally continue execution.
}

// If we made it here, the language needs to be written.
```